### PR TITLE
Moved description of methods to traits

### PR DESCRIPTION
**Pull request by xgreenx**
Moved description of methods to traits
Added short description of each trait.
Added `data` argument to transfer function(which can be passed to `on_received`

### CODE REVIEW

File: PSPs/psp-17.md [hunk 1]
```diff
@@ -36,51 +36,113 @@ The main motivation for this proposal is to have one **trait** that shares the s
...
-
+ /// Transfers `value` amount of tokens from the caller's account to account `to`
+ /// with additional `data` with no specified format.
+ /// Emits a `Transfer` event on success.
+ /// 
+ /// This method also calls [on_received](#on_received) method on `to`.
+ /// 
+ /// **Errors**
+ /// * Panics with `InsufficientBalance` error if there are not enough tokens on
+ /// the caller's account Balance.
```
Comment (VargSupercolony) [inline PSPs/psp-17.md]: balance should be lowercased, it refers to literal balance, not the type in ink!.

File: PSPs/psp-17.md [hunk 2]
```diff
@@ -36,51 +36,113 @@ The main motivation for this proposal is to have one **trait** that shares the s
...
+ /// Allows `spender` to withdraw from the caller's account multiple times, up to the `value` amount.
+ /// If this function is called again it overwrites the current allowance with `value`.
+ /// Emits `Approval` event.
+ /// 
+ /// **Errors**
+ /// * Panics with `ZeroSenderAddress` error if sender's address is zero.
+ /// * Panics with `ZeroRecipientAddress` error if recipient's address is zero.
  fn approve(&mut self, spender: AccountId, value: Balance);
 
+ /// Atomically increases the allowance granted to `spender` by the caller on `delta_value`.
```
Comment (VargSupercolony) [inline PSPs/psp-17.md]: "on" should be changed to "by" ("by `delta_value`")

File: PSPs/psp-17.md [hunk 3]
```diff
@@ -36,51 +36,113 @@ The main motivation for this proposal is to have one **trait** that shares the s
...
+ /// Emits `Approval` event.
+ /// 
+ /// **Errors**
+ /// * Panics with `ZeroSenderAddress` error if sender's address is zero.
+ /// * Panics with `ZeroRecipientAddress` error if recipient's address is zero.
  fn increase_allowance(&mut self, spender: AccountId, delta_value: Balance);
 
+ /// Atomically decreases the allowance granted to `spender` by the caller on `delta_value`.
```
Comment (VargSupercolony) [inline PSPs/psp-17.md]: same here, on -> by

**Review by VargSupercolony — CHANGES_REQUESTED** [review]
great overall, just minor mistakes

**Review by VargSupercolony — APPROVED** [review]
moved description of methods to trait

File: PSPs/psp-17.md [hunk 4]
```diff
@@ -36,51 +36,113 @@ The main motivation for this proposal is to have one **trait** that shares the s
...
-type AccountId = [u8; 32];
-// u128 must be enough to cover most of the use cases of standard token.
-type Balance = u128;
-```
+Another suggestion is to have `PSP17Receiver` and call `on_received` method at the end of transfer if the recipient is a contract.
 
 ### Traits
 
 ```rust
+/// PSP17 is a trait of Fungible Token Standard
```
Comment (sventime) [inline PSPs/psp-17.md:48]: Add a .

**Review by sventime — COMMENTED** [review]
Let's rename Unknown error to Custom

**Review by 0xMarkian — APPROVED** [review]

**Pull request merged**